LinqToDBPostGisNetTopologySuite.Tests/LinearReferencingTests.cs

```diff
@@ -0,0 +1,318 @@
+using System.Linq;
+
+using LinqToDB;
+using NUnit.Framework;
+using NTSG = NetTopologySuite.Geometries;
+
+namespace LinqToDBPostGisNetTopologySuite.Tests
+{
+    [TestFixture]
+    class LinearReferencingTests : TestsBase
+    {
+        [SetUp]
+        public void Setup()
+        {
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                db.TestGeometries.Delete();
+            }
+        }
+
+        [Test]
+        public void TestSTLineInterpolatePoint()
+        {
+            var geomText = "LINESTRING(25 50, 100 125, 150 190)";
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                db.TestGeometries
+                .Value(g => g.Id, 1)
+                .Value(g => g.Geometry, () => GeometryInput.STGeomFromEWKT(geomText))
+                .Insert();
+
+                var result = db.TestGeometries
+                .Where(g => g.Id == 1)
+                .Select(g => g.Geometry.STLineInterpolatePoint(0.20))
+                .Single() as NTSG.Point;
+
+                Assert.AreEqual(51.5974135047432, result.X, 1.0E-8);
```

Comment:
Be careful to copy these values from PostGIS Reference Examples; it sometimes differs from own results (may be due another PostgreSql / PostGIS versions). I'll think, how to use more obvious values in tests.

Reply:
Some simple test cases with short expects (integer woubld be better )may be helpful?

Reply:
Not sure about integers, but test case with `STLineInterpolatePoint(0.5)` and asserting like this
``c#
Assert.AreEqual((100-25)/2, result.X, 1.0E-8);
``
may look better